homeassistant/components/yessssms/notify.py

```diff
@@ -3,39 +3,68 @@
 
 import voluptuous as vol
 
+from YesssSMS import YesssSMS
+
 from homeassistant.const import CONF_PASSWORD, CONF_RECIPIENT, CONF_USERNAME
 import homeassistant.helpers.config_validation as cv
 
 from homeassistant.components.notify import PLATFORM_SCHEMA, BaseNotificationService
 
+
+from .const import CONF_PROVIDER
+
 _LOGGER = logging.getLogger(__name__)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Required(CONF_USERNAME): cv.string,
         vol.Required(CONF_PASSWORD): cv.string,
         vol.Required(CONF_RECIPIENT): cv.string,
+        vol.Optional(CONF_PROVIDER, default="YESSS"): cv.string,
     }
 )
 
 
 def get_service(hass, config, discovery_info=None):
     """Get the YesssSMS notification service."""
-    return YesssSMSNotificationService(
-        config[CONF_USERNAME], config[CONF_PASSWORD], config[CONF_RECIPIENT]
+
+    try:
+        yesss = YesssSMS(
+            config[CONF_USERNAME], config[CONF_PASSWORD], provider=config[CONF_PROVIDER]
+        )
+    except YesssSMS.UnsupportedProviderError as ex:
+        _LOGGER.error("Unknown provider: %s", ex)
+        return None
+    try:
+        if yesss.login_data_valid():
```

Comment:
We can invert this check, log error if true and return None. Then we can outdent the logging on success. Combine that message with the debug message below.

Reply:
you can't. logging login data as valid would be false in the case of an offline HA instance.

I however inverted the check and moved the valid message out a little, but withing the try block.
we could move that message in an else belonging to the except.

Reply:
I don't follow. We already have the case I want first below in the else.